lua/harpoon/init.lua

```diff
@@ -146,6 +146,7 @@ function Harpoon.setup(self, partial_config)
 
     ---@diagnostic disable-next-line: param-type-mismatch
     self.config = Config.merge_config(partial_config, self.config)
+    self.data = Data.Data:new(self.config)
```

Comment:
Found this MR just before creating my pull request :) Looks like it takes time to get it merged.
One suggestion from me would be to check at least for the `partial_config` to be non-empty. Otherwise we don't need to reload data and  calling `sync_on_change` (e.g. if it was just harpoon.setup({})).
```suggestion
    if partial_config then
        self.data = Data.Data:new(self.config)
    end
```
(and the same `if` before calling `sync_on_change`).